pkg/sanity/sanity.go

```diff
@@ -123,11 +134,28 @@ func (sc *SanityContext) setup() {
 	}
 
 	By("creating mount and staging directories")
-	err = createMountTargetLocation(sc.Config.TargetPath)
-	Expect(err).NotTo(HaveOccurred())
-	if len(sc.Config.StagingPath) > 0 {
-		err = createMountTargetLocation(sc.Config.StagingPath)
+	// If callback function for creating target dir is specified, use it.
+	if sc.Config.CreateTargetDir != nil {
+		newpath, err := sc.Config.CreateTargetDir(sc.Config.TargetPath)
+		Expect(err).NotTo(HaveOccurred())
+		sc.targetPath = newpath
+	} else {
+		err = createMountTargetLocation(sc.Config.TargetPath)
 		Expect(err).NotTo(HaveOccurred())
+		sc.targetPath = sc.Config.TargetPath
+	}
+
+	if len(sc.Config.StagingPath) > 0 {
+		// If callback function for creating staging dir is specified, use it.
+		if sc.Config.CreateStagingDir != nil {
+			newpath, err := sc.Config.CreateStagingDir(sc.Config.StagingPath)
+			Expect(err).NotTo(HaveOccurred())
+			sc.stagingPath = newpath
```

Comment:
Always explain what the error is about, see https://github.com/kubernetes/kubernetes/issues/34059. I know, the rest of the code doesn't do that either (would be a good enhancement), but it's never too late to do it better.

This could is rather repetitive. I would change the `createMountTargetLocation` method and then use it like this:

```
sc.targetPath = createMountTargetLocation(sc.Config.TargetPath, sc.Config.CreateTargetDir)
sc.stagingPath = createMountTargetLocation(sc.Config.StagingPath, sc.Config.CreateStagingDir)
```

Speaking of `createMountTargetLocation`: there is a call to it in `node.go` which causes directories to be created on the host even when `sc.Config.StagingPath` is set. I think that call can be removed.

Reply:
I can confirm that in scenario we use (staging and target directories are created on VMs where pods run, host acting only as test orchestrator) I see /var/tmp/tmp.XXXXXXXXXX directories created on host which are not used and remain undeleted, both with csi-test 1.0.2 and with current PR change.
Positive feedback is that tests pass same way with current PR changes (as with 1.0.2)